File: internal/schema/temporary_employee.go
package schema


type CreateTemporaryEmployeeRequest struct {
	FirstName      string `json:"firstName" validate:"required"`
	LastName       string `json:"lastName" validate:"required"`
	Department     string `json:"department"`
	Position       string `json:"position"`
	ContactNumber  string `json:"contactNumber"`
	OfficeLocation string `json:"officeLocation"`
	IsActive       bool   `json:"isActive"`
}

type UpdateTemporaryEmployeeRequest struct {
	FirstName      string `json:"firstName"`
	LastName       string `json:"lastName"`
	Department     string `json:"department"`
	Position       string `json:"position"`
	ContactNumber  string `json:"contactNumber"`
	OfficeLocation string `json:"officeLocation"`
	IsActive       *bool  `json:"isActive"`
}

type SearchCriteria struct {
	FirstName      string `query:"firstName"`
	LastName       string `query:"lastName"`
	Department     string `query:"department"`
	Position       string `query:"position"`
	OfficeLocation string `query:"officeLocation"`
	IsActive       *bool  `query:"isActive"`
}

File: ui/types/temporaryEmployee.ts
export interface TemporaryEmployee {
  ID: number
  FirstName: string
  LastName: string
  Department: string
  Position: string
  ContactNumber: string
  OfficeLocation: string
  IsActive: boolean
  CreatedAt: string
  UpdatedAt: string
  DeletedAt: string | null
}

export interface CreateTemporaryEmployeeRequest {
  FirstName: string
  LastName: string
  Department?: string
  Position?: string
  ContactNumber?: string
  OfficeLocation?: string
  IsActive?: boolean
}

export interface UpdateTemporaryEmployeeRequest {
  FirstName?: string
  LastName?: string
  Department?: string
  Position?: string
  ContactNumber?: string
  OfficeLocation?: string
  IsActive?: boolean
}

export interface SearchCriteria {
  FirstName?: string
  LastName?: string
  Department?: string
  Position?: string
  OfficeLocation?: string
  IsActive?: boolean
}